clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1796,6 +1824,26 @@ void SYCLIntegrationHeader::emitForwardClassDecls(
         }
         break;
       }
+      case TemplateArgument::ArgKind::Integral: {
+        // Handle Kernel Name Type templated using enum.
+        QualType T = Arg.getIntegralType();
+        if (const EnumType *ET = T->getAs<EnumType>()) {
+          auto EnumDecl = ET->getDecl();
+          DeclContext *DC = EnumDecl->getDeclContext();
```

Comment:
The decl context isn't necessarily a namespace, what do we do in the class case?

Reply:
The check for emitting the forward declaration below is -  if (!NS || (NS && !InCLNameSpace)). So if it isn't a namespace we would just call emitFwdDecl. The namespace check was only because I thought we should add the inCLNameSpace check. I'm removing the check now based on review comments